R/gsplot-class.R

```diff
@@ -27,11 +35,27 @@ gsplot.default <- function(..., created=Sys.Date(),
                            gsplot.version=packageDescription(getPackageName(), 
                                                              fields = "Version"),
                            config.file=NA, theme=NA, frame.plot=TRUE) {
-  object <- gsplot(list(metadata=list(created=created,
-                                      gsplot.version=gsplot.version),
-                        global=list('config'=list(frame.plot=frame.plot, 
-                                                  config.file=!is.na(config.file),
-                                                  theme=!is.na(theme)))))
+  
+  if(!all(is.na(theme))){
+
+    if(!all(is.na(config.file))){
+      config.file <- theme$global$config.file
+    }
+    
+    object <- gsplot(c(list(metadata=list(created=created,
+                                        gsplot.version=gsplot.version),
+                            global=c(list('config'=list(frame.plot=theme[["global"]][["config"]][["frame.plot"]], #this means you can't override the theme's frame.plot. You could use box() later though
+                                                    config.file=!is.na(config.file))),
+                                      theme[["global"]][names(theme[["global"]])[names(theme[["global"]]) != "config"]])),
+                            theme[names(theme)[!(names(theme) %in% c("metadata","global"))]]
+                          )
+                     )
+  } else {
+    object <- gsplot(list(metadata=list(created=created,
+                                        gsplot.version=gsplot.version),
+                          global=list('config'=list(frame.plot=frame.plot, 
+                                                    config.file=!is.na(config.file)))))    
+  }
```

Comment:
rather than set global and metadata in two places, create theme object OR gsplot object THEN add global/metadata